catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/security/accesscontrol/TagBaseVisitorTest.java

```diff
@@ -0,0 +1,292 @@
+  public void testDisjoint() {
+    verifyMethodCalled(ff.disjoint(TEST, new PointImpl()));
+  }
+
+  @Test
+  public void testDWithin() {
+    verifyMethodCalled(ff.dwithin(TEST, new PointImpl(), 10.0, ""));
+  }
+
+  @Test
+  public void testEquals() {
+    verifyMethodCalled(ff.equals(TEST, new PointImpl()));
+  }
+
+  @Test
+  public void testIntersects() {
+    verifyMethodCalled(ff.intersects(TEST, new PointImpl()));
+  }
+
+  @Test
+  public void testOverlaps() {
+    verifyMethodCalled(ff.overlaps(TEST, new PointImpl()));
+  }
+
+  @Test
+  public void testTouches() {
+    verifyMethodCalled(ff.touches(TEST, new PointImpl()));
+  }
+
+  @Test
+  public void testWithin() {
+    verifyMethodCalled(ff.within(TEST, new PointImpl()));
+  }
+
+  @Test
+  public void testNullFilter() {
+    baseVisitor.visitNullFilter(new Object());
+    assertThat(baseVisitor.isMethodCalled(), is(true));
+  }
+
+  @Test
+  public void testAnyInteracts() {
+    verifyMethodCalled(ff.anyInteracts(ff.property(TEST), ff.literal(TEST)));
+  }
+
+  @Test
+  public void testAfter() {
+    verifyMethodCalled(ff.after(ff.property(TEST), ff.literal(TEST)));
+  }
+
+  @Test
+  public void testBefore() {
+    verifyMethodCalled(ff.before(ff.property(TEST), ff.literal(TEST)));
+  }
+
+  @Test
+  public void testBegins() {
+    verifyMethodCalled(ff.begins(ff.property(TEST), ff.literal(TEST)));
+  }
+
+  @Test
+  public void testBegunBy() {
+    verifyMethodCalled(ff.begunBy(ff.property(TEST), ff.literal(TEST)));
+  }
+
+  @Test
+  public void testDuring() {
+    verifyMethodCalled(ff.during(ff.property(TEST), ff.literal(TEST)));
+  }
+
+  @Test
+  public void testEndedBy() {
+    verifyMethodCalled(ff.endedBy(ff.property(TEST), ff.literal(TEST)));
+  }
+
+  @Test
+  public void testEnds() {
+    verifyMethodCalled(ff.ends(ff.property(TEST), ff.literal(TEST)));
+  }
+
+  @Test
+  public void testMeets() {
+    verifyMethodCalled(ff.meets(ff.property(TEST), ff.literal(TEST)));
+  }
+
+  @Test
+  public void testMetBy() {
+    verifyMethodCalled(ff.metBy(ff.property(TEST), ff.literal(TEST)));
+  }
+
+  @Test
+  public void testOverlappedBy() {
+    verifyMethodCalled(ff.overlappedBy(ff.property(TEST), ff.literal(TEST)));
+  }
+
+  @Test
+  public void testTContains() {
+    verifyMethodCalled(ff.tcontains(ff.property(TEST), ff.literal(TEST)));
+  }
+
+  @Test
+  public void testTEquals() {
+    verifyMethodCalled(ff.tequals(ff.property(TEST), ff.literal(TEST)));
+  }
+
+  @Test
+  public void testTOverlaps() {
+    verifyMethodCalled(ff.toverlaps(ff.property(TEST), ff.literal(TEST)));
+  }
+
+  private void verifyMethodCalled(Filter filter) {
+    filter.accept(baseVisitor, null);
+    assertThat(baseVisitor.isMethodCalled(), is(true));
+  }
+
+  private void verifyMethodIgnored(Filter filter) {
+    filter.accept(baseVisitor, null);
+    assertThat(baseVisitor.isMethodCalled(), is(false));
+  }
+
+  private static class TagBaseVisitorUnderTest extends TagBaseVisitor {
```

Comment:
✏️ Alternatively, you could make your visitor a `Spy` and use `verify`.

Reply:
Ah that will cut down on some code. I will explore that.

Reply:
Going to leave this be. The difference is negligible because I'd still need the inner concrete class defined upon which to spy, which is most of the boilerplate anyway.